packages/react/src/components/TableCard/TableCard.jsx

```diff
@@ -386,7 +386,14 @@ const TableCard = ({
           isSortable: true,
           width: i.width ? `${i.width}px` : newSize === CARD_SIZES.LARGETHIN ? '150px' : '', // force the text wrap
           filter: i.filter
-            ? i.filter
+            ? i.type === 'TIMESTAMP'
+              ? {
+                  filterFunction: (cellValue, filterValue) => {
```

Comment:
We should spread the other filter options into this object. Additionally, we need to check if a filterFunction was supplied. We wouldn't want to replace it with this default if the devs want to filter differently.